src/workers/cdc.ts

```diff
@@ -165,6 +172,10 @@ const onUserChange = async (
     ) {
       await notifyDevCardEligible(logger, data.payload.after.id);
     }
+
+    if (data.payload.after.reputation >= submissionAccessThreshold) {
```

Comment:
I just noticed that this will always get published after the user reaches the threshold. There should be another check that the before value is less than the threshold. Please add tests accordingly.

Reply:
Totally agree. Now that we need a persistent value for this case. What do you think about adding a property in our `alerts` table (or anywhere else that fits) for checking if an email was sent already?

Reply:
Ah, yes. But I've also thought of an edge case where toggling the upvote of someone's comment that would make them pass the threshold would re-send the email, again and again. Should we cover that as well? Or is it something we can pass on?

Reply:
Ah, yes. We shouldn't send that unnecessary property. Now that I've checked SendGrid's API, and they don't have anything that would allow us to check an email attempt programmatically, the option I am thinking of now is at the user's table.

Like for DevCard, we will create `sentSubmissionEligibility`. What do you think?

Reply:
Great idea. I think we can create an `email_events` (or `single_email_events`) table - from there we will have the user id (PK) and template id as another column. If the said template id was stored, then that means, it was sent already. What do you think?

Reply:
First of all, don't think it has to do only with emails. It can be potentially anything. At this point, we don't add anything to the email other than the first name. So I wouldn't hassle to think about payloads. It can even be something like UserState entity. It doesn't need to store the template id. Just user id, key, and value. According to the key, you should know how to proceed. In our case, it will be emailed, but it can be any other thing as well